api/s2n.h

```diff
@@ -1269,6 +1269,21 @@ S2N_API extern int s2n_config_set_ticket_decrypt_key_lifetime(struct s2n_config
  */
 S2N_API extern int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *name, uint32_t name_len,
         uint8_t *key, uint32_t key_len, uint64_t intro_time_in_seconds_from_epoch);
+
+/**
+ * Gates ticket-based resumption to TLS 1.3, as the secret used in TLS 1.2 resumption
+ * is not forward-secret. Clients should not expect session tickets from servers and
+ * servers will not send session tickets when TLS 1.2 is negotiated and forward secrecy
+ * is enforced. 
+ * 
+ * @note The default behavior is that forward secrecy is not enforced
+ *
+ * @param config The config object being updated
+ * @param enabled Indicates if forward secrecy is enforced or not on tickets
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure
+ */
+S2N_API extern int s2n_config_enforce_ticket_forward_secrecy(struct s2n_config *config, bool enabled);
+
```

Comment:
Idk about "enforce". It sounds like more of an active effort than just not using tickets. Maybe "require" instead? I don't feel strongly about this though.